benchmarks/benchmark_comparison.jl

```diff
@@ -0,0 +1,280 @@
+                HagerZhangLinesearch(M; sigma=0.5),
+                mem_len,
+                gtol;
+                retraction_method=ProjectionRetraction(),
+                vector_transport_method=ProjectionTransport(),
+            )
+
+            push!(times_manopt, manopt_time)
+            println("Manopt.jl time: $(manopt_time) ms")
+            println("Manopt.jl iterations: $(manopt_iters)")
+            println("Manopt.jl objective: $(manopt_obj)")
+
+            optim_time, optim_iters, optim_obj = benchmark_time_state(
+                OptimQN(), M, N, f, g!, x0, ls_hz, mem_len, gtol
+            )
+            println("Optim.jl  time: $(optim_time) ms")
+            push!(times_optim, optim_time)
+            println("Optim.jl  iterations: $(optim_iters)")
+            println("Optim.jl  objective: $(optim_obj)")
+        end
+        plot!(
+            plt,
+            N_vals,
+            times_manopt;
+            label="Manopt.jl ($manifold_name)",
+            xaxis=:log,
+            yaxis=:log,
+        )
+        plot!(
+            plt,
+            N_vals,
+            times_optim;
+            label="Optim.jl ($manifold_name)",
+            xaxis=:log,
+            yaxis=:log,
+        )
+    end
+    xticks!(plt, N_vals, string.(N_vals))
+
+    return plt
+end
+
+# generate_cmp(N -> (f_rosenbrock, g_rosenbrock!), mem_len=4)
+
+function generate_rayleigh_problem(N::Int)
+    A = Symmetric(randn(N, N) / N)
+    f_manopt = ManoptExamples.RayleighQuotientCost(A)
+    g_manopt! = ManoptExamples.RayleighQuotientGrad!!(A)
+    M = Manifolds.Sphere(N - 1)
+    f_ret(x) = f_manopt(M, x)
+    f_ret(M::AbstractManifold, x) = f_manopt(M, x)
+    g_ret!(storage, x) = g_manopt!(M, storage, x)
+    g_ret!(M::AbstractManifold, storage, x) = g_manopt!(M, storage, x)
+    return (f_ret, g_ret!)
+end
+# generate_cmp(generate_rayleigh_problem, manifold_names=[:Sphere], mem_len=4, N_vals=[10, 100, 1000])
+
+function test_case_manopt()
+    N = 4000
+    mem_len = 4
+    M = Manifolds.Sphere(N - 1; parameter=:field)
+    ls_hz = LineSearches.HagerZhang()
+
+    x0 = zeros(N)
+    x0[1] = 1
+    manopt_sc = StopWhenGradientNormLess(1e-6; norm=norm_inf) | StopAfterIteration(1000)
+
+    return quasi_Newton(
+        M,
+        f_rosenbrock,
+        g_rosenbrock!,
+        x0;
+        #stepsize=Manopt.LineSearchesStepsize(ls_hz),
+        stepsize=HagerZhangLinesearch(M),
+        evaluation=InplaceEvaluation(),
+        vector_transport_method=ProjectionTransport(),
+        return_state=true,
+        memory_size=mem_len,
+        stopping_criterion=manopt_sc,
+    )
+end
+
+function test_case_optim()
+    N = 4
+    mem_len = 2
+    ls_hz = LineSearches.HagerZhang()
+    method_optim = LBFGS(; m=mem_len, linesearch=ls_hz, manifold=Optim.Flat())
+    options_optim = Optim.Options(; g_tol=1e-6)
+
+    x0 = zeros(N)
+    x0[1] = 0
+    optim_state = optimize(f_rosenbrock, g_rosenbrock!, x0, method_optim, options_optim)
```

Comment:
Similar to the previous all, this could also be one line just the `optimize` call whose return value is returned?